### PR TITLE
Add all files for my project(might be some trash)

### PR DESCRIPTION
**Pull request by kapatejlb**
In libraries entities, in web application no entities.

### CODE REVIEW

File: Data.Entities/Entities/Commentary.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        public int Id { get; set; }
+        public string Text { get; set; }
+
+        public DateTime Date { get; set; }
+
+        public int ProjectId { get; set; }
+        public Project Project { get; set; }
+
+
```
Comment (monomarh) [inline Data.Entities/Entities/Commentary.cs]: unnecessary line break

File: Data.Entities/Entities/Project.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        public ICollection<Commentary> Comments { get; set; }
+
+        public Content Content { get; set; }
+
+        public ICollection<Post> Posts {get; set;}
+
+        public string UserId { get; set; }
+        public AspNetUsers AspNetUsers { get; set; }
+
+
```
Comment (monomarh) [inline Data.Entities/Entities/Project.cs]: unnecessary line break

File: Data.Entities/Entities/Project.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using Data.Entities.Entities;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Data.Entities
+{
+    public class Project
+    {
+        public Project()
```
Comment (monomarh) [inline Data.Entities/Entities/Project.cs]: unnecessary constructor

File: Data.EntitiyFrameWork/DataBase1Context.cs [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+using Data.Entities;
+using Data.Entities.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata;
+
+namespace Data.EntitiyFrameWork
+{
+    public partial class DataBase1Context : DbContext
+    {
+        public DataBase1Context()
```
Comment (monomarh) [inline Data.EntitiyFrameWork/DataBase1Context.cs:11]: unnecessary constructor

File: Data.EntitiyFrameWork/DataBase1Context.cs [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        public virtual DbSet<Project> Projects { get; set; }
+
+        public virtual DbSet<Post> Posts { get; set; }
+
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+                optionsBuilder.UseSqlServer("Server=(localdb)\\mssqllocaldb;Database=DataBase;Trusted_Connection=True;");
```
Comment (monomarh) [inline Data.EntitiyFrameWork/DataBase1Context.cs]: delete this comment

File: Data.EntitiyFrameWork/DataBase1Context.cs [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+                entity.Property(e => e.NormalizedUserName).HasMaxLength(256);
+
+                entity.Property(e => e.UserName).HasMaxLength(256);
+            });
+
+            modelBuilder.Entity<Project>(entity =>
+            {
+                entity
+                .HasOne(d => d.Content)
```
Comment (monomarh) [inline Data.EntitiyFrameWork/DataBase1Context.cs]: indent codestyles

File: Web/Controllers/ProjectController.cs [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+using Microsoft.AspNetCore.Mvc;
+
+
+namespace Web.Controllers
+{
+    public class ProjectController : Controller
+    {
+        private readonly IProjectService projectService;
+
+        //ivate readonly
```
Comment (monomarh) [inline Web/Controllers/ProjectController.cs]: delete in this file unnecessary comments

File: Web/Controllers/ProjectController.cs [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+            var content = projectService.GetContent(id);
+            project.Content = content;
+
+            return View(project);
+        }
+
+        [HttpPost]
+        public ActionResult Details(int idofproj, string NewCommentary, string UserName)
+        {
+
```
Comment (monomarh) [inline Web/Controllers/ProjectController.cs]: line breaks

File: Web/Views/Home/Privacy.cshtml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+<!DOCTYPE HTML>
+<html>
+<head>
+    <meta charset="utf-8">
+    <title>Тег TEXTAREA</title>
+</head>
+<body>
+
+    <form action="textarea1.php" method="post">
```
Comment (monomarh) [inline Web/Views/Home/Privacy.cshtml]: php in c# project?????!!!!!!

File: Web/Views/Project/Create.cshtml [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+<hr />
+<div class="row">
+    <div class="col-md-12">
+        <form asp-action="Create">
+            <div asp-validation-summary="ModelOnly" class="text-danger"></div>
+
+
+            @*<input asp-for="Id" hidden="" class="form-control" />*@
+
+            <div class="form-group" style="width: 500px;">
```
Comment (monomarh) [inline Web/Views/Project/Create.cshtml:16]: might be better if you move this style in css

File: Web/wwwroot/css/site.css [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  text-align: center;
+  word-break: break-all;
+}
+
+/* Sticky footer styles
+-------------------------------------------------- */
+html {
+  font-size: 14px;
+}
+@media (min-width: 768px) {
```
Comment (monomarh) [inline Web/wwwroot/css/site.css:16]: add line break

File: Web/wwwroot/js/colorchange.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
+﻿var a = true;
```
Comment (monomarh) [inline Web/wwwroot/js/colorchange.js]: rename variable, for example: isCustomBackgroundColor

**Review by monomarh — CHANGES_REQUESTED** [review]

File: Data.Entities/Entities/AspNetRoleClaims.cs [hunk 13]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (ilya007900) [inline Data.Entities/Entities/AspNetRoleClaims.cs]: unused usings

File: Data.EntitiyFrameWork/DataBase1Context.cs (hunk 5 above)
Comment (ilya007900) [inline Data.EntitiyFrameWork/DataBase1Context.cs]: Connection string should be in appsettings.json. Try to remove this method.

File: Data.Entities/Entities/AspNetUserClaims.cs [hunk 14]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (ilya007900) [inline Data.Entities/Entities/AspNetUserClaims.cs]: unused usings

File: Data.Entities/Entities/Commentary.cs [hunk 15]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (ilya007900) [inline Data.Entities/Entities/Commentary.cs]: unused usings

File: Data.Entities/Entities/Commentary.cs [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+
+        public DateTime Date { get; set; }
+
+        public int ProjectId { get; set; }
+        public Project Project { get; set; }
+
+
+        public string UserId { get; set; }
+        public AspNetUsers aspNetUser {get; set;}
+
```
Comment (ilya007900) [inline Data.Entities/Entities/Commentary.cs]: Correct name is just User. And you can remove UserName and use User.Name

File: Data.Entities/IManageEntities.cs [hunk 17]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Data.Entities
+{
+    public interface IManageEntities<T>
```
Comment (ilya007900) [inline Data.Entities/IManageEntities.cs]: Better name is IRepository

File: Data.EntitiyFrameWork/DataBase1Context.cs [hunk 18]
```diff
@@ -0,0 +1,172 @@
...
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+                optionsBuilder.UseSqlServer("Server=(localdb)\\mssqllocaldb;Database=DataBase;Trusted_Connection=True;");
+            }
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.HasAnnotation("ProductVersion", "2.2.6-servicing-10079");
```
Comment (ilya007900) [inline Data.EntitiyFrameWork/DataBase1Context.cs]: Better delete this line

File: Data.EntitiyFrameWork/Repository.cs [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+using System.Text;
+
+
+namespace Data.EntitiyFrameWork
+{
+    public class Repository<T> : IManageEntities<T> where T : class
+    {
+        private readonly DataBase1Context _DbContext;
+
+        public Repository(DataBase1Context DbContext)
```
Comment (ilya007900) [inline Data.EntitiyFrameWork/Repository.cs]: name method parameters with camel case (dbContext)

File: Data.Service/ProjectService.cs [hunk 20]
```diff
@@ -0,0 +1,149 @@
...
+        }
+        public int Add(ProjectViewModel Entity, string UserName)
+        {
+            var project = new Project();
+            project.Title = Entity.Title;
+            project.Created = Entity.Created;
+            project.Subject = Entity.Subject;
+            project.ExpirationDate = Entity.ExpirationDate;
+
+            var userName = dbContext.AspNetUsers.Where(a => a.UserName == UserName).ToList()[0];
```
Comment (ilya007900) [inline Data.Service/ProjectService.cs]: Don't use [0]. Use First() or FirstOrDefault() linq methods.

File: Data.Service/ProjectService.cs [hunk 21]
```diff
@@ -0,0 +1,149 @@
...
+    public class ProjectService : IProjectService
+    {
+        private readonly DataBase1Context dbContext;
+        public ProjectService(DataBase1Context dbContext)
+        {
+            this.dbContext = dbContext;
+        }
+        public int Add(ProjectViewModel Entity, string UserName)
+        {
+            var project = new Project();
```
Comment (ilya007900) [inline Data.Service/ProjectService.cs]: Use object initializer. (https://www.c-sharpcorner.com/UploadFile/mahesh/object-initializer-in-C-Sharp/)

File: Data.Service/ProjectService.cs [hunk 22]
```diff
@@ -0,0 +1,149 @@
...
+            project.Title = Entity.Title;
+            project.Created = Entity.Created;
+            project.Subject = Entity.Subject;
+            project.ExpirationDate = Entity.ExpirationDate;
+
```
Comment (ilya007900) [inline Data.Service/ProjectService.cs]: Use object initializer (https://www.c-sharpcorner.com/UploadFile/mahesh/object-initializer-in-C-Sharp/)

File: Data.Service/ProjectService.cs (hunk 20 above)
Comment (ilya007900) [inline Data.Service/ProjectService.cs]: Don't use [0]. Use First() or FirstOrDefault() linq methods.

File: Data.Service/ProjectService.cs [hunk 23]
```diff
@@ -0,0 +1,149 @@
...
+            dbContext.Projects.Add(project);
+            dbContext.SaveChanges();
+            return project.Id;
+        }
+
+        public void AddCommentary(int Id, string CommentaryText, string UserName)
+        {
+
+            var commentary = new Commentary();
```
Comment (ilya007900) [inline Data.Service/ProjectService.cs]: Use object initializer (https://www.c-sharpcorner.com/UploadFile/mahesh/object-initializer-in-C-Sharp/)

File: Data.Service/ProjectService.cs [hunk 24]
```diff
@@ -0,0 +1,149 @@
...
+            
+            commentary.Text = CommentaryText;
+            commentary.ProjectId = Id;
+            commentary.Date = DateTime.Now;
+            commentary.UserName = UserName;
+
+            var userName = dbContext.AspNetUsers.Where(a => a.UserName == UserName).ToList()[0];
```
Comment (ilya007900) [inline Data.Service/ProjectService.cs]: Don't use [0]. Use First() or FirstOrDefault() linq methods.

File: Data.Service/ProjectService.cs [hunk 25]
```diff
@@ -0,0 +1,149 @@
...
+            entity.Comments = new List<Commentary>();
+            entity.Comments.Add(commentary);
+
+            dbContext.Commentaries.Add(commentary);
+            dbContext.SaveChanges();
+        }
+
+        public void AddContent(int id, string Content, string UserName)
+        {
+            var content = new Content();
```
Comment (ilya007900) [inline Data.Service/ProjectService.cs]: Use object initializer (https://www.c-sharpcorner.com/UploadFile/mahesh/object-initializer-in-C-Sharp/)

File: Data.Service/ProjectService.cs [hunk 26]
```diff
@@ -0,0 +1,149 @@
...
+
+            content.Date = DateTime.Now;
+            content.ProjectId = id;
+            content.Text = Content;
+
+            var userName = dbContext.AspNetUsers.Where(a => a.UserName == UserName).ToList()[0];
```
Comment (ilya007900) [inline Data.Service/ProjectService.cs]: Don't use [0].

File: Data.Service/ProjectService.cs [hunk 27]
```diff
@@ -0,0 +1,149 @@
...
+            var entity = dbContext.Projects.Find(id);
+            content.Project = entity;
+
+
+
+            entity.Content = new Content();
```
Comment (ilya007900) [inline Data.Service/ProjectService.cs]: Use object initializer.

File: Data.Service/ProjectService.cs [hunk 28]
```diff
@@ -0,0 +1,149 @@
...
+        {
+            var entity = dbContext.Projects.Find(Id);
+            dbContext.Projects.Remove(entity);
+            dbContext.SaveChanges();
+        }
+
+        public ProjectViewModel Get(int Id)
+        {
+            var projectViewModel = new ProjectViewModel();
+            var project = dbContext.Projects.Find(Id);
```
Comment (ilya007900) [inline Data.Service/ProjectService.cs:112]: Use object initializer.

File: Web/Program.cs [hunk 29]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
```
Comment (ilya007900) [inline Web/Program.cs]: unused usings

File: Web/Views/Project/Create.cshtml [hunk 30]
```diff
@@ -0,0 +1,62 @@
+﻿@model Data.ViewModels.ViewModels.ProjectViewModel
+
+@{
+    ViewData["Title"] = "Create";
+    var AllowedCategories = new List<string>() { "Arts", "Comics & Illustration", "Design & Tech", "Film", "Food & Craft", "Games", "Publishing" };
```
Comment (ilya007900) [inline Web/Views/Project/Create.cshtml]: Hard code. Better is create static list in ProjectViewModel and initialize it.

File: Web/Views/Project/Create.cshtml [hunk 31]
```diff
@@ -0,0 +1,62 @@
...
+            <div class="form-group">
+                <input asp-for="ExpirationDate" class="form-control" />
+                <span asp-validation-for="ExpirationDate" class="text-danger"></span>
+            </div>
+
+            <div class="form-group">
+                <input type="submit" value="Create" class="btn btn-primary" />
+            </div>
+
+            <input hidden name ="UserName" value="@User.Identity.Name" ></input>
```
Comment (ilya007900) [inline Web/Views/Project/Create.cshtml]: input tag don't have close tag.

File: Web/Views/Project/Edit.cshtml [hunk 32]
```diff
@@ -0,0 +1,57 @@
+﻿@model Data.ViewModels.ViewModels.ProjectViewModel
+
+@{
+    ViewData["Title"] = "Edit";
+    var AllowedCategories = new List<string>() { "Arts", "Comics & Illustration", "Design & Tech", "Film", "Food & Craft", "Games", "Publishing" };
```
Comment (ilya007900) [inline Web/Views/Project/Edit.cshtml]: Hard code. Better is create static list in ProjectViewModel and initialize it.

**Pull request closed without merging**